lib/objectid.js

```diff
@@ -0,0 +1,12 @@
+var ObjectId;
+
+try {
+  ObjectId = require('mongodb').ObjectId;
+} catch(e) {
+  if (e.code === 'MODULE_NOT_FOUND')
+    throw e
+}
+
```

Comment:
To be honest, I don't like adding `mongodb` dependency to juggler at all. Remember, there is no guarantee that `mongodb` module referenced by juggler will be the same module as referenced by `loopback-connector-mongodb`. In fact, this will be the most frequent case.

```
app
  node_modules
    loopback-datasource-juggler
      mongodb
    loopback-connector-mongodb
      mongodb
```

As a result, you are registering a type (constructor function) that is actually never used by the connector, because the connector has it's own local type (constructor function).